pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -100,7 +100,7 @@ while [ "$#" -gt 0 ]; do
         shift 1
         ;;
       --use-remote-sudo)
-        maybeSudo=(sudo --)
+        maybeSudo=(sudo --preserve-env=NIXOS_INSTALL_BOOTLOADER --)
```

Comment:
perhaps something like this in case we add more vars?
```
# comma separated list of vars to preserve when using sudo
preservedSudoVars="NIXOS_INSTALL_BOOTLOADER"
```

```
        maybeSudo=(sudo  --preserve-env=$preservedSudoVars --)
```

Reply:
Done. As an aside passing state via envvars through nixos-rebuild -> switch-to-configuration -> systemd-boot-builder seems a bit error prone. Is there a better way to fix the way these three scripts are coordinating for this feature? With some direction I'd be happy to tackle a change like that as a separate PR.

Reply:
The env var should always be preserved so users can do it without the flag as seen here https://nixos.wiki/wiki/Bootloader#Re-installing_the_bootloader

Reply:
I missed the variable already being documented as an API.

Reply:
Done.